apis/v1alpha1/generator.yaml

```diff
@@ -63,9 +68,13 @@ resources:
           operation: ChangeResourceRecordSets
           path: ChangeBatch.Changes.ResourceRecordSet.GeoLocation
       HealthCheckId:
-        from:
-          operation: ChangeResourceRecordSets
-          path: ChangeBatch.Changes.ResourceRecordSet.HealthCheckId
+        # from:
+        #   operation: ChangeResourceRecordSets
+        #   path: ChangeBatch.Changes.ResourceRecordSet.HealthCheckId
+        references:
+          resource: HealthCheck
+          path: Status.ID
+          is_immutable: true
```

Comment:
I don't know if it is ok to change this here without having breaking changes on current deployments.
It currently just removes the reference from the CRD which is not ok, it should keep it but with as HealthCheckId in the crd with a HealthCheckRef as well, like for the hostedZoneId. But the generator does not do this.

Reply:
Maybe I'll have to do this in a second PR for the changes to be taken into account by the generator?

Reply:
I see your concern and I agree it's something we should avoid doing now... (removing existing fields). Ideally we need to extend the code generator to handle these kind of cases.

Reply:
> Maybe I'll have to do this in a second PR for the changes to be taken into account by the generator?

That would be awesome! Feel free to work on it, and ping me on k8s slack if you need a code walk through of the code generator internals :)

Reply:
All right then we agree, I will remove this change in this PR.